### ISSUE TITLE
User Story: Allow Nano Servers to be automatically joined to an AD Domain

### ISSUE THREAD

**Issue by PlagueHO** [PlagueHO/LabBuilder#85]
At the moment, AD Domain joining is performed using DSC and the xActiveDirectory DSC resource.

This resource is dependent on Windows Features that are not available on Nano Server.

To join a Nano Server to a Domain will most likely require a different method.

**Comment by MS-MatthewWalker** [comment]
Will have to use DJoin to create a blob and inject that into the VM, that means testing until the domain is available  - at least that is the only way I can think of today

**Comment by PlagueHO** [comment]
Yeah, I thought that might have been the only way - that is how I was doing it in the New-NanoServerVHD script I wrote. That is going to require some thought.

It actually could be possible to upload a script to the Nano Server that causes it to remote connect to the DC and execute the DJOIN command to generate the blob, download the blob and then use it to join itself to the domain. That would eliminate the host from being a go-between to transfer the blob and execute the calls. However, the big problem with that is that the Nano Server would need to have the DC credentials written to this "script" - which is just bad behavior. So I think that I'll rule that out.

**Comment by PlagueHO** [comment]
I might look at raising a PR with the xComputerName resource to allow it to use DJOIN blob files to join to a domain. This will allow us to implement the Nano DJOIN without introducing a completely new process (although it will be different - but still using DSC).

**Comment by PlagueHO** [comment]
This is currently blocked and waiting for the xActiveDirectory xADComputer resource to be completed and have ODJoin support added to it.

**Comment by PlagueHO** [comment]
The work on the DSC Resources is complete so this should be able to be completed.
